src/Internal/DependencyManagement/ExtendedContainer.php

```diff
@@ -0,0 +1,102 @@
+<?php
+/**
+ * Class ExtendedContainer
+ *
+ * @package WooCommerce\Payments
+ */
+
+namespace WCPay\Internal\DependencyManagement;
+
+use WCPay\Vendor\League\Container\Container;
+use WCPay\Vendor\League\Container\Definition\Definition;
+
+/**
+ * Extends the League container to allow WCPay customizations.
+ *
+ * During tests, `wcpay_get_test_container()` will return an instance of this class.
+ */
+class ExtendedContainer extends Container {
```

Comment:
It is not really exposed to devs because the extended container is a private property of `WCPay\Container`.  `WCPay\Container` does not, and it should never expose those methods of the internal container.

I thought about having a separate container for tests, to avoid shipping those few methods with the plugin, but achieving that would involve hacks. Seems like the easiest and most straight-forward method is to use reflection during tests, and access the private property.

By the way, the main `WCPay\Container` class is still the one that's supplied to classes, and also the one responsible for loading service providers. I don't want to include any code, which allows access to/replacement of the internal `WCPay\Internal\DependencyManagement\ExtendedContainer`.